src/plugins/intel_cpu/src/utils/shape_inference/shape_inference_cpu.hpp

```diff
@@ -18,6 +20,11 @@ class IShapeInfer {
 public:
```

Comment:
Maybe this interface can inherit from `IShapeInferCommon` (or some new Iface which got common members for both interfaces) or vice versa?

Reply:
Unfortunately, we had to implement this interface, since the `IShapeInferCommon` use incompatible static shape representation and conversations take too much time. I mean `VectorDims` <-> `StaticShape`, map of host tensors <-> map of memory object pointers. So to implement optimized shape inference operations, optimized here means CPU plugin specifics and constraints, we introduced this interface, fully compatible with the plugin, and just wrapped `IShapeInferCommon` as more generic. But, I see that nobody else use `IShapeInferCommon` so m.b. the entire subsytem may adopt the CPU plugin specific `IShapeInfer `, but under the hood we will have to convert `VectorDims` <-> `StaticShape` and input maps anyway.